zrml/swaps/src/benchmarks.rs

```diff
@@ -395,57 +475,133 @@ benchmarks! {
             Some(a.saturated_into()),
             None,
             ScoringRule::CPMM,
-            false
+            false,
+            None,
         );
         let _ = Pallet::<T>::mutate_pool(pool_id, |pool| {
-            pool.pool_status = PoolStatus::Closed; Ok(())
+            pool.pool_status = PoolStatus::Closed;
+            Ok(())
         });
     }: {
-        Pallet::<T>::clean_up_pool(
-            &MarketType::Categorical(a as u16),
-            pool_id, &OutcomeReport::Categorical(0),
-            &account("ScrapCollector", 0, 0)
+        Pallet::<T>::clean_up_pool_categorical(
+            pool_id,
+            &OutcomeReport::Categorical(0),
+            &account("ScrapCollector", 0, 0),
         )?;
     }
 
     swap_exact_amount_in_cpmm {
-        let a = T::MaxAssets::get();
+        // We're trying to get as many iterations in `bpow_approx` as possible. Experiments have
+        // shown that y = 3/4, weight_ratio=1/2 (almost) maximizes the number of iterations for
+        // calculating y^r within the set of values allowed in `swap_exact_amount_in` (see
+        // `math::calc_out_given_in`). To get these values, we use the following parameters:
+        // amount_in = 1/3 * balance_in, weight_in = 1, weight_out = 2.
+        let asset_count = T::MaxAssets::get();
```

Comment:
Using this value probably overcharges the users a bit in most cases, because it assumes during pool encoding that we'll have the maximum amount of `assets` and `weights`. I think though, that the value we overcharge in this case is so low, that we can neglect it here and save the extra time required for the benchmark to determine it.

Reply:
I think the difference was surprisingly large, like 10%. I believe Chris and I had a discussion about this. The reason seems to be that the vector needs to be deserialized, so it's `O(len)`. Nevertheless, I argued against parametrizing the benchmark by the asset count, since that would mean basically parametrizing every extrinsic which reads a pool.

Definitely something to keep in mind, though.

Reply:
I think it makes sense to do that for structures that can grow quite big. I am fine with leaving it out now, we over-estimate which is fine and we can respect decoding in a future optimization run.